### PR TITLE
mysql (ticdc): back port batch dml to release 6.1

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/7653

### What is changed and how it works?
This PR is a manual cherry pick of https://github.com/pingcap/tiflow/pull/7548.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/sqlmodel/multivalue.go [hunk 1]
```diff
@@ -200,3 +206,143 @@ func GenInsertSQL(tp DMLType, changes ...*RowChange) (string, []interface{}) {
 	}
 	return buf.String(), args
 }
+
+// GenUpdateSQL generates the UPDATE SQL and its arguments.
+// Input `changes` should have same target table and same columns for WHERE
+// (typically same PK/NOT NULL UK), otherwise the behaviour is undefined.
+// Compared to GenInsertSQL with DMLInsertOnDuplicateUpdate, this function is
+// slower and more complex, we should only use it when PK/UK is updated.
+func GenUpdateSQL(changes ...*RowChange) (string, []interface{}) {
```
Comment (lance6716) [inline pkg/sqlmodel/multivalue.go:215]: lgtm for this function

**Review by lance6716 — APPROVED** [review]
lgtm for GenUpdateSQL

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**